defrag/modules/bugs.py

```diff
@@ -0,0 +1,184 @@
+# Defrag - centralized API for the openSUSE Infrastructure
+# Copyright (C) 2021 openSUSE contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+from typing import Any, Dict, List, Tuple
+from defrag import app, LOGGER
+from defrag.modules.helpers import QueryObject
+from defrag.modules.helpers.sync_utils import as_async
+from defrag.modules.helpers.exceptions import BugzillaException, ParsingException, NetworkException
+import bugzilla
+from bugzilla import BugzillaError
+from xmlrpc.client import Fault
+from fastapi import Response, status
+import aiohttp
+from bs4 import BeautifulSoup
+import re
+
+
+URL = "bugzilla.opensuse.org"
+bzapi = bugzilla.Bugzilla(URL)
+
+__MOD_NAME__ = "bugs"
+
+
+@app.get("/" + __MOD_NAME__ + "/")
+async def root() -> Dict[str, str]:
+    return {
+        "module": "Bugzilla",
+        "description": "Get information about bugs on bugzilla.opensuse.org"}
+
+
+@as_async
+def get_bug_from_bugzilla(query: QueryObject) -> Dict[str, Any]:
+    id = int(query.context["id"])
+    try:
+        bug = bzapi.getbug(id)
+    except BugzillaError as exp:
+        LOGGER.error(
+            f"Error occured while getting bug from Bugzilla: {exp.get_bugzilla_error_string} (Error code {exp.get_bugzilla_error_code})")
+        raise BugzillaException(
+            exp.get_bugzilla_error_string +
+            f"({exp.get_bugzilla_error_code})")
+    result = {"id": id}
+    result["product"] = bug.product
+    result["component"] = bug.component
+    result["status"] = bug.status
+    result["resolution"] = bug.resolution
+    result["summary"] = bug.summary
+    result["url"] = f"https://{URL}/show_bug.cgi?id={id}"
+    return result
+
+
+@app.get("/" + __MOD_NAME__ + "/bug/")
+async def get_bug(id: int, response: Response) -> Dict[str, str]:
```

Comment:
then you are able to be more specific in the return type:
```python
async def get_bug(id: int, respnse: Response) -> BugZillaEntry
```

Reply:
Done